landoscript/src/landoscript/lando.py

```diff
@@ -0,0 +1,92 @@
+import asyncio
+import datetime
+import logging
+from pprint import pprint
+from typing import Any, Callable
+
+from aiohttp import ClientResponseError, ClientSession
+from async_timeout import timeout
+from scriptworker.utils import calculate_sleep_time, retry_async
+
+from landoscript.errors import LandoscriptError
+
+log = logging.getLogger(__name__)
+
+
+LandoAction = dict[str, str]
+
+
+def create_commit_action(commitmsg: str, diff: str) -> LandoAction:
+    """Return a `create-commit` lando action. Primarily exists to centralize the author name."""
+    author = "Release Engineering Landoscript <release+landoscript@mozilla.com>"
+    timestamp = datetime.datetime.now(datetime.timezone.utc).isoformat()
+
+    return {"action": "create-commit", "commitmsg": commitmsg, "diff": diff, "date": timestamp, "author": author}
+
+
+async def submit(
+    session: ClientSession,
+    lando_api: str,
+    lando_token: str,
+    lando_repo: str,
+    actions: list[LandoAction],
+    sleeptime_callback: Callable[..., Any] = calculate_sleep_time,
+) -> str:
+    """Submit the provided `actions` to the given `lando_repo` through the `lando_api`."""
+    url = f"{lando_api}/api/v1/{lando_repo}"
+    json = {"actions": actions}
+
+    log.info(f"submitting actions to lando: {actions}")
+    async with timeout(30):
+        log.info(f"submitting POST request to {url}")
+        log.info("message body is:")
+        log.info(pprint(json))
+
+        submit_resp = await retry_async(
+            session.post,
+            args=(url,),
+            kwargs={
+                "json": json,
+                "raise_for_status": True,
+                # TODO: is this a bearer token?
+                "headers": {"Authorization": f"Bearer {lando_token}"},
```

Comment:
You will need to add a `User-Agent` here that looks like `Lando-User/{username}`.

Reply:
Can I use whatever value I want here?

Reply:
Any value is fine, but something like the email or username for the bot would be preferred. :+1: